lib/codecs/src/decoding/format/protobuf.rs

```diff
@@ -0,0 +1,311 @@
+    fn from(config: &ProtobufDeserializerConfig) -> Self {
+        Self::new(config.desc_file.clone(), config.message_type.clone())
+    }
+}
+
+fn to_vrl(
+    prost_reflect_value: prost_reflect::Value,
+    kind: &prost_reflect::Kind,
+) -> vector_common::Result<vrl::value::Value> {
+    let v = match prost_reflect_value {
+        prost_reflect::Value::Bool(v) => vrl::value::Value::from(v),
+        prost_reflect::Value::I32(v) => vrl::value::Value::from(v),
+        prost_reflect::Value::I64(v) => vrl::value::Value::from(v),
+        prost_reflect::Value::U32(v) => vrl::value::Value::from(v),
+        prost_reflect::Value::U64(v) => vrl::value::Value::from(v),
+        prost_reflect::Value::F32(v) => vrl::value::Value::Float(
+            NotNan::new(f64::from(v)).map_err(|_e| format!("Float number cannot be Nan"))?,
+        ),
+        prost_reflect::Value::F64(v) => vrl::value::Value::Float(
+            NotNan::new(v).map_err(|_e| format!("F64 number cannot be Nan"))?,
+        ),
+        prost_reflect::Value::String(v) => vrl::value::Value::from(v),
+        prost_reflect::Value::Bytes(v) => vrl::value::Value::from(v),
+        prost_reflect::Value::EnumNumber(v) => {
+            let enum_desc = kind.as_enum().unwrap();
+            vrl::value::Value::from(
+                enum_desc
+                    .get_value(v)
+                    .ok_or_else(|| format!("The number {} cannot be in {}", v, enum_desc.name()))?
+                    .name(),
+            )
+        }
+        prost_reflect::Value::Message(mut v) => {
+            let mut obj_map = BTreeMap::new();
+            for field_desc in v.descriptor().fields() {
+                let field = v.get_field_mut(&field_desc);
+                let mut taken_value = prost_reflect::Value::Bool(false);
+                std::mem::swap(&mut taken_value, field);
+                let out = to_vrl(taken_value, &field_desc.kind())?;
+                obj_map.insert(field_desc.name().to_string(), out);
+            }
+            vrl::value::Value::from(obj_map)
+        }
+        prost_reflect::Value::List(v) => {
+            let vec = v
+                .into_iter()
+                .map(|o| to_vrl(o, &kind))
+                .collect::<Result<Vec<_>, vector_common::Error>>()?;
+            vrl::value::Value::from(vec)
+        }
+        prost_reflect::Value::Map(v) => {
+            let message_desc = kind.as_message().unwrap();
+            vrl::value::Value::from(
+                v.into_iter()
+                    // TODO: handle unwrap
+                    .map(|kv| {
+                        (
+                            kv.0.as_str().unwrap().to_string(),
+                            to_vrl(kv.1, &message_desc.map_entry_value_field().kind()).unwrap(),
+                        )
+                    })
+                    .collect::<BTreeMap<String, _>>(),
+            )
+        }
+    };
+    Ok(v)
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
To avoid hardcoding paths, please add a helper like the following: 
```rust
fn test_data_dir() -> PathBuf {
    PathBuf::from(env::var_os("CARGO_MANIFEST_DIR").unwrap()).join("tests/data/decoding/protobuf")
}

```

Nit: Feel free to skip this, but I prefer `"decoding/protobuf"` to  `"protobuf_decoding"`.